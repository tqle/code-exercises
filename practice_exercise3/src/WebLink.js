import React from "react";
import { validateWebSite } from "./utils/linkUtil";

/**
 * Web Link extends link tag with
 * custom title to indicate link is an web url.
 */
class WebLink extends React.Component {
  render() {
    // Destructuring objects from props.
    let { website } = this.props;

    // Check for empty, undefined, null cases
    let validWebSite = validateWebSite(website);
    // FUTURE: Write a regex or pattern match if website is valid url format
    let title = validWebSite ? "URL: " + validWebSite : "";
    let linkReference = validWebSite || "#";

    return (
      <a title={title} href={linkReference}>
        {validWebSite}
      </a>
    );
  }
}

export default WebLink;
