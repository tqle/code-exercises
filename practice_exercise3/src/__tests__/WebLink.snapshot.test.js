import React from "react";
import { create } from "react-test-renderer";
import WebLink from "../WebLink";

/**
 * Test Web Link snapshot
 */
test("<WebLink /> renders correctly", () => {
  const test = create(<WebLink website="kale.biz" />);
  expect(test.toJSON()).toMatchSnapshot();
});

test("<WebLink /> renders as an anchor when no web site reference is set", () => {
  const test = create(<WebLink>Kale.biz</WebLink>).toJSON();
  expect(test).toMatchSnapshot();
});
