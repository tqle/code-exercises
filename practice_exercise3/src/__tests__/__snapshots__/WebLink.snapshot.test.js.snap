// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebLink /> renders as an anchor when no web site reference is set 1`] = `
<a
  href="#"
  title=""
>
  
</a>
`;

exports[`<WebLink /> renders correctly 1`] = `
<a
  href="kale.biz"
  title="URL: kale.biz"
>
  kale.biz
</a>
`;
